controllers/operands/wasp.go

```diff
@@ -0,0 +1,225 @@
+		overwritten := false
+		if req.HCOTriggered {
+			req.Logger.Info("Updating existing wasp DaemonSet Spec to new opinionated values")
+		} else {
+			req.Logger.Info("Reconciling an externally updated wasp DaemonSet Spec to its opinionated values")
+			overwritten = true
+		}
+		hcoutil.MergeLabels(&wasp.ObjectMeta, &found.ObjectMeta)
+		wasp.Spec.DeepCopyInto(&found.Spec)
+		err := Client.Update(req.Ctx, found)
+		if err != nil {
+			return false, false, err
+		}
+		return true, overwritten, nil
+	}
+
+	return false, false, nil
+}
+
+func (*waspHooks) justBeforeComplete(_ *common.HcoRequest) { /* no implementation */ }
+
+func (*waspHooks) getConditions(cr runtime.Object) []metav1.Condition {
+	return dsConditionsToK8s(cr.(*appsv1.DaemonSet).Status.Conditions)
+}
+
+func NewWasp(hc *hcov1beta1.HyperConverged) *appsv1.DaemonSet {
+	waspImage, _ := os.LookupEnv(hcoutil.WaspImageEnvV)
+	wasp := NewWaspWithNameOnly(hc)
+	wasp.Spec = appsv1.DaemonSetSpec{
+		Selector: &metav1.LabelSelector{
+			MatchLabels: map[string]string{
+				"name": "wasp",
+			},
+		},
+		UpdateStrategy: appsv1.DaemonSetUpdateStrategy{
+			Type: appsv1.RollingUpdateDaemonSetStrategyType,
+			RollingUpdate: &appsv1.RollingUpdateDaemonSet{
+				MaxUnavailable: ptr.To(intstr.FromString("10%")),
+				MaxSurge:       ptr.To(intstr.FromString("0")),
+			},
+		},
+		Template: corev1.PodTemplateSpec{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels: map[string]string{
+					"name": "wasp",
+				},
+			},
+			Spec: corev1.PodSpec{
+				RestartPolicy:                 corev1.RestartPolicyAlways,
+				ServiceAccountName:            waspServiceAccount,
+				HostPID:                       true,
+				HostUsers:                     ptr.To(true),
+				TerminationGracePeriodSeconds: ptr.To(int64(30)),
+				Containers: []corev1.Container{
+					{
+						Name:            "wasp-agent",
+						Image:           waspImage,
+						ImagePullPolicy: corev1.PullAlways,
+						Env: []corev1.EnvVar{
+							{
+								Name:  "FSROOT",
+								Value: "/host",
+							},
+							{
+								Name:  "SWAPINNES",
+								Value: "5",
+							},
+							{
+								Name:  "LOOP",
+								Value: "true",
+							},
+							{
+								Name:  "DEBUG",
+								Value: "",
+							},
+							{
+								Name:  "DRY",
+								Value: "",
+							},
+						},
+						Resources: corev1.ResourceRequirements{
+							Requests: corev1.ResourceList{
+								corev1.ResourceCPU:    resource.MustParse("100m"),
+								corev1.ResourceMemory: resource.MustParse("50M"),
+							},
+						},
+						SecurityContext: &corev1.SecurityContext{
+							Privileged: ptr.To(true),
+						},
+						VolumeMounts: []corev1.VolumeMount{
+							{
+								Name:      "host",
+								MountPath: "/host",
+							},
+						},
+					},
+				},
+				Volumes: []corev1.Volume{
+					{
+						Name: "host",
+						VolumeSource: corev1.VolumeSource{
+							HostPath: &corev1.HostPathVolumeSource{
+								Path: "/",
```

Comment:
is it really required to mount all of the host's filesystem? can it be done with the relevant path(s)?